pkg/plugin/driver/daemonset/daemonset.go

```diff
@@ -78,19 +85,46 @@ func (p *Plugin) GetResultType() string {
 	return p.Definition.ResultType
 }
 
+//FillTemplate populates the internal Job YAML template with the values for this particular job.
+func (p *Plugin) FillTemplate(hostname string) (*bytes.Buffer, error) {
+	var b bytes.Buffer
+	// TODO (EKF): Should be YAML once we figure that out
+	container, err := utils.ContainerToJSON(&p.Definition.Spec)
+	if err != nil {
+		return &bytes.Buffer{}, errors.Wrapf(err, "couldn't reserialize container for daemonset %q", p.Definition.Name)
+	}
+
+	vars := templateData{
+		PluginName:        p.Definition.Name,
+		ResultType:        p.Definition.ResultType,
+		SessionID:         p.SessionID,
+		Namespace:         p.Namespace,
+		ProducerContainer: container,
+		MasterAddress:     getMasterAddress(hostname), // TODO(EKF)
+	}
+
+	if err := daemonSetTemplate.Execute(&b, vars); err != nil {
+		return &bytes.Buffer{}, errors.Wrapf(err, "couldn't fill template %q", p.Definition.Name)
+	}
+	return &b, nil
+}
+
 // Run dispatches worker pods according to the DaemonSet's configuration.
-func (p *Plugin) Run(kubeclient kubernetes.Interface) error {
+func (p *Plugin) Run(kubeclient kubernetes.Interface, hostname string) error {
 	var (
-		b         bytes.Buffer
 		daemonSet v1beta1ext.DaemonSet
 	)
-	p.Definition.Template.Execute(&b, p.DfnTemplateData)
+	b, err := p.FillTemplate(hostname)
+	if err != nil {
+		// Already wrapped sufficiently by FillTemplate
+		return err
```

Comment:
nit: it might be nice to know explicitly that FillTemplate failed vs deducing it from the final error we see wrapped up. This would also let you delete the comment

Also applies to job plugin driver